### PR TITLE
Add getResamplingIndices()

### PR DESCRIPTION
**Pull request by pat-s**
### New function: `getResamplingIndices()`

Partly solves https://github.com/mlr-org/mlr/issues/2266

1. Retrieve the resampling indices of any `ResampleResult` object
2. Using `inner = TRUE`, one can retrieve the inner indices if `resample()` was called using `extract = getTuneResult`

Full example: https://github.com/mlr-org/mlr/compare/inner-resamp-inds?expand=1#diff-fbcca53cfe214295c23946d97c143420

```r
getResamplingIndices(r, inner = TRUE)

[[1]]
[[1]]$train.inds
[[1]]$train.inds[[1]]
 [1] 72 73  5 18 20 64 27 19 12 32 26  9  2 39 74 40 44 61 34  6 50 41 71 23 75 63 17 51  8 31 69  7 56 22 29 11 49 35 54 21 66 25 67 38 60 65 14 33 59 70


[[1]]$test.inds
[[1]]$test.inds[[1]]
 [1] 58 62 45 43 28  1 37 10  3 15 13 53 36 16 55 30 42  4 68 48 46 24 47 52 57



[[2]]
[[2]]$train.inds
[[2]]$train.inds[[1]]
 [1] 54 16  4 67 28 48 60 73 26 40 17 22 33 50 15  8 64 75 57 63 66 53 70 19 27 51 74 14 36 72 68 52 55 65 42 30  1 71 69 39 12  2 56 24 61 34 43 32 11 58


[[2]]$test.inds
[[2]]$test.inds[[1]]
 [1]  5 18  3  7 47 20 10 25 13 45 21 38 35 46 31  6 41  9 59 29 62 49 44 37 23
```

PS: I changed my role from "ctb" to "aut" and added my Orcid. If that's not okay, just tell me and I roll it back :)

### CODE REVIEW

**Comment by larskotthoff** [comment]
Please don't edit NEWS directly, put the changes in the [wiki](https://github.com/mlr-org/mlr/wiki/NEWS). It looks like a bunch of unrelated changes are in this PR.

**Comment by pat-s** [comment]
Adding NEWS to the wiki is new to me - might have missed that discussion.

I reverted the NEWS and fixed two indentations. Other than that all changes are required.

**Comment by larskotthoff** [comment]
This functionality is not specific to tuning, is it? Could we have a generic implementation that works for different types of results?

**Comment by pat-s** [comment]
@larskotthoff changed the function name to  `getResamplingIndices`. It works now for both the outer and inner level using the switch `inner = TRUE` (default FALSE).

**Comment by larskotthoff** [comment]
Great, thanks. It still feels like this function should work for any kind of nested resampling, not just for tuning.

**Comment by pat-s** [comment]
Ah I wasn't aware that we have more "inner" methods. It now supports also `getFeatSelResult`. Is there another kind that I am missing?

**Comment by larskotthoff** [comment]
Thanks, that should be it. Could you update the documentation and add a test please?

**Comment by pat-s** [comment]
@larskotthoff ready?

File: R/getResamplingIndices.R [hunk 1]
```diff
@@ -0,0 +1,29 @@
+#' @title Get the resampling indices from a nested tuning.
```
Comment (larskotthoff) [inline R/getResamplingIndices.R]: Documentation still refers to only tuning.

File: R/getResamplingIndices.R [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+#'
+#' @description
+#' After you resampled a tuning wrapper (see [makeTuneWrapper])
```
Comment (larskotthoff) [inline R/getResamplingIndices.R]: See above.

File: R/getResamplingIndices.R [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+#' with `resample(..., extract = getTuneResult)` or `resample(..., extract = getFeatSelResult)` this helper returns a `list` with
+#' the resampling indices used for the respective method.
+#'
+#' @param object ([ResampleResult]) \cr
+#'   The result of resampling of a tuning wrapper.
```
Comment (larskotthoff) [inline R/getResamplingIndices.R]: See above.

File: R/getResamplingIndices.R [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#' @param inner ([logical]) \cr
+#'   If `TRUE`, returns the inner indices of a nested resampling setting.
+#' @return ([list]). One list for each outer resampling fold.
+#' @family tune
+#' @examples
+#' # see example of makeTuneWrapper
```
Comment (larskotthoff) [inline R/getResamplingIndices.R]: An example here that shows how to use `getResamplingIndices()` would be great.

**Comment by larskotthoff** [comment]
Thanks, merging.

**Pull request merged**